### PR TITLE
Adding router

### PR DESCRIPTION
**Pull request by Jhanemba64**

### CODE REVIEW

**Review by akornne — APPROVED** [review]

**Review by PierreA44 — APPROVED** [review]

**Review by bayram-69 — APPROVED** [review]

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -1,19 +1,15 @@
...
 function App() {
   const [basketCount, setBasketCount] = useState(0);
   return (
     <div className="App">
       <Navbar basketCount={basketCount} />
-      <div className="main">
-        <Filtres />
-        <CardList basketCount={basketCount} setBasketCount={setBasketCount} />
-      </div>
+      <Outlet setBasketCount={setBasketCount} />
```
Comment (ioayoub) [inline frontend/src/App.jsx:12]: Never pass a props to the Outlet (unless you really know what you're doing) , your App.jsx is not designed to handle states.

File: frontend/src/components/CardList/CardList.jsx [hunk 2]
```diff
@@ -13,21 +14,24 @@ function CardList({ basketCount, setBasketCount }) {
...
-            smallImage={p.images.small}
-            largeImage={p.images.large}
-            price={p.cardmarket.prices.averageSellPrice}
-            types={p.types}
-            basketCount={basketCount}
-            setBasketCount={setBasketCount}
-          />
-        ))}
+    <div className={styles.search}>
+      <Filtres />
```
Comment (ioayoub) [inline frontend/src/components/CardList/CardList.jsx:18]: All component's names must be in english

File: frontend/src/main.jsx [hunk 3]
```diff
@@ -1,14 +1,48 @@
...
+        loader: () => {
+          const cards = fetch(
+            "https://api.pokemontcg.io/v2/cards?pageSize=50"
+          ).then((res) => res.json());
+          return defer({ cards });
+        },
+        children: [
+          {
+            path: "/search/:id",
+            element: <CardPage />,
```
Comment (ioayoub) [inline frontend/src/main.jsx:33]: You add directly the loader here to fetch your datas, unless I missed something?

**Review by ioayoub — CHANGES_REQUESTED** [review]
Some required changes are commented.
Avoid also using empty components (ex: Homepage) they're useless here.

**Review by ioayoub — APPROVED** [review]

**Pull request merged**